pkg/syncer/syncer.go

```diff
@@ -17,54 +17,162 @@ limitations under the License.
 package syncer
 
 import (
+	"bytes"
 	"context"
 	"fmt"
+	"io"
 	"strings"
+	"time"
 
 	elementalv1 "github.com/rancher/elemental-operator/api/v1beta1"
+	"github.com/rancher/elemental-operator/pkg/log"
 	"github.com/rancher/elemental-operator/pkg/object"
+
+	upgradev1 "github.com/rancher/system-upgrade-controller/pkg/apis/upgrade.cattle.io/v1"
+
+	corev1 "k8s.io/api/core/v1"
 	"k8s.io/client-go/kubernetes"
-	"k8s.io/client-go/rest"
-	"sigs.k8s.io/controller-runtime/pkg/client"
+	ctrl "sigs.k8s.io/controller-runtime"
 )
 
 const (
-	jsonType   = "json"
-	customType = "custom"
+	jsonType        = "json"
+	customType      = "custom"
+	defaultMountDir = "/data"
```

Comment:
The /data/output path is fixed here because we use an opinionated path as the channel image output.
For a moment it took me a while to understand who wrote the file...
I know that this has just been moved from the custom.go code, so nothing new, but maybe we can leverage we are moving the code to add a short comment here to help to better follow the flow.

More in general, my guess is that we don't need a volume anymore as the volume is used inside the new pod only, but we use it to stay compatible with current channel image behavior. Did I got it or am I missing something?